src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -98,10 +102,13 @@ public sealed override object ReadInline(object item, IEdmTypeReference edmType,
                 throw Error.Argument("item", SRResources.ArgumentMustBeOfType, typeof(ODataResource).Name);
             }
 
+            // Create the appropriate nested context
+            ODataDeserializerContext nestedContext = BuildNestedContextFromCurrentContext(readContext, structuredType.IsEntity() ? ApplyIdToPath(readContext, resourceWrapper) : readContext.Path);
```

Comment:
>BuildNestedContextFromCurrentContext [](http://example.com/codeflow?start=53&length=36)

why do we need to build the nested context at the beginning?

It could be a regression?

Reply:
This was a bug. The nested context that was being used was the incorrect one. That's why this was added to ensure that the right nested context is used.